2015/proposals/akshay.md

```diff
@@ -0,0 +1,125 @@
+# Add taxonomic name resolution to the EcoData Retriever to facilitate data intensive approaches to ecology.
+
+## Abstract
+
+The EcoData Retriever is an engine which automates the tasks of finding,
+downloading, and cleaning up ecological data files, and then stores them in a
+local database of your choice. The program cleans and munges the datasets and
+formats them  before inserting it into your database.
+
+This project will tackle the problem of resolving scientific names stored
+inconsistently across different datasets. This would allow the users to more
+easily combine different datasets and answer scientific questions which was not
+possible with the use of single datasets alone.
+
+## Technical Details
+There are three main components in the project:
+
+* Accessing and incorporating the resolution service into the retriever
+
+* Building the control flow logic for querying
+
+* Updating the user interfaces
+
+**Accessing and incorporating the resolution service into the retriever**:
+
+The main component of this project would be to build a system to resolve a
+species name by sending it to one of many taxonomic name resolution services like
+[Global Names Resolver](http://resolver.globalnames.org/) which is already supported by pytaxize.
+and processing the result to determine the best name. This can be done by
+incorporating the feature in the retriever directly or using the already
+existing library [pytaxize](https://github.com/sckott/pytaxize) in the retriever.
+I would propose to use pytaxize and use that in the retriever as this will
+ensure modularity and help develop pytaxize in the process so that it becomes a
+complete port of its R counterpart [taxize](https://github.com/ropensci/taxize).
+This will involve a fair amount of work as different name resolution services
+have to be tested by running variety of data and have to compared for the
+results. And a fair amount of thought has to be given for cases wherein two
+different services give two equally good results.
+
+**Building the control flow logic for querying**:
+
+The second component of this project would be to restrict the running of
+resolution depending on the type of data. This can be done by adding a separate
+column in the scripts so that it contains the species name. We will make sure
+that the resolution will not run in the absence of the species name information in the table.
+
+**Updating the user interfaces**:
+
+The third component involves updating the user interface of the retriever to
+felicitate users to perform Taxonomic name resolution.
```

Comment:
You might add a few details about what the UI for integrating taxonomic name resolution would look like and how it would fit into the existing Retriever. What types of options would users have?